db/db_basic_test.cc

```diff
@@ -4361,6 +4361,39 @@ TEST_P(DBBasicTestDeadline, IteratorDeadline) {
   Close();
 }
 
+#ifndef ROCKSDB_LITE
+TEST_F(DBBasicTest, GetCompressedWalsAfterSync) {
+  Options options = GetDefaultOptions();
+  options.wal_recovery_mode = WALRecoveryMode::kPointInTimeRecovery;
+  options.create_if_missing = true;
+  options.env = env_;
+  options.avoid_flush_during_recovery = true;
+  options.track_and_verify_wals_in_manifest = true;
+  // Enable WAL compression so that the newly-created WAL will be non-empty
+  // after DB open, even if point-in-time WAL recovery encounters no
+  // corruption.
+  options.wal_compression = kZSTD;
+  DestroyAndReopen(options);
```

Comment:
On certain platforms without ZSTD capable of stream compression, maybe do something like
```cpp
if (db_->GetOptions().wal_compression == kNone) {
  ROCKSDB_TEST_BYPASS("stream compression not present");
  return;
}
```

Reply:
Ah ok. Thanks for the tip.